scripts/CMakeLists.txt

```diff
@@ -67,7 +67,7 @@ if(ONEDAL_DPCTL_INTEGRATION)
     message(STATUS "DPCTL_INCLUDE_DIR:" ${DPCTL_INCLUDE_DIR})
 endif()
 
-if(ONEDAL_DIST_SPMD)
+if(IFACE STREQUAL "spmd_dpc")

```

Comment:
I'd prefer to eval `IFACE STREQUAL "spmd_dpc"` elsewhere and store the result in a variable, e.g. `IFACE_IS_SPMD_DPC`.

Reply:
I don't want mess up a lot this branch  with a refactoring a lot.
I can probably do this as a follow up

Reply:
You can eval it once on top and re-use the variable in the remaining file. I don't like the repeated string comparisons. One typo and everything goes south!

Reply:
it was removed in this PR no? isn't it passed from build_backend script and set using the same logic that is used here?

Reply:
I don't want to duplicate logic. If we are calling make files for building SPMD backend, that is already means that we are going to build onedal spmd backend. So there is no need for additional flag there.